plugins/org.eclipse.papyrus.uml.interaction.model/src/org/eclipse/papyrus/uml/interaction/internal/model/commands/NudgeOnRemovalCommand.java

```diff
@@ -0,0 +1,313 @@
+/*****************************************************************************
+ * Copyright (c) 2018 Johannes Faltermeier and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Johannes Faltermeier - Initial API and implementation
+ *****************************************************************************/
+package org.eclipse.papyrus.uml.interaction.internal.model.commands;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.emf.common.command.Command;
+import org.eclipse.emf.common.command.CommandWrapper;
+import org.eclipse.emf.common.command.IdentityCommand;
+import org.eclipse.emf.edit.domain.EditingDomain;
+import org.eclipse.gmf.runtime.notation.View;
+import org.eclipse.papyrus.uml.interaction.internal.model.impl.LogicalModelPlugin;
+import org.eclipse.papyrus.uml.interaction.internal.model.impl.MElementImpl;
+import org.eclipse.papyrus.uml.interaction.model.MElement;
+import org.eclipse.papyrus.uml.interaction.model.MExecution;
+import org.eclipse.papyrus.uml.interaction.model.MInteraction;
+import org.eclipse.papyrus.uml.interaction.model.MLifeline;
+import org.eclipse.papyrus.uml.interaction.model.MMessage;
+import org.eclipse.papyrus.uml.interaction.model.MOccurrence;
+import org.eclipse.papyrus.uml.interaction.model.spi.DiagramHelper;
+import org.eclipse.papyrus.uml.interaction.model.spi.LayoutHelper;
+import org.eclipse.uml2.uml.Element;
+
+/**
+ * This command analyses the current graph and fills the white space created by the deletion of elements.
+ */
+public class NudgeOnRemovalCommand extends CommandWrapper {
+
+	private static final int DEFAULT_VERTICAL_OFFSET = 10;
+
+	private EditingDomain editingDomain;
+
+	private Set<MElement<? extends Element>> mElementsToRemove;
+
+	private MInteraction interaction;
+
+	private Collection<Element> elementsToRemove;
+
+	/**
+	 * Creates a new {@link NudgeCommand}.
+	 * 
+	 * @param editingDomain
+	 *            the {@link EditingDomain}
+	 * @param interaction
+	 *            the {@link MInteraction}
+	 * @param elementsToRemove
+	 *            the elements which will be removed
+	 */
+	public NudgeOnRemovalCommand(EditingDomain editingDomain, MInteraction interaction,
+			Collection<Element> elementsToRemove) {
+
+		this.editingDomain = editingDomain;
+		this.interaction = interaction;
+		this.elementsToRemove = elementsToRemove;
```

Comment:
`elementsToRemove` doesn't need to be a field because it's only used to calculate `mElementsToRemove`.  So it can just be a parameter of the `getRemovedMElements` method.